conanfile.py

```diff
@@ -84,15 +86,29 @@ class Xrpl(ConanFile):
         'xxhash/*:shared': False,
     }
 
+    def validate(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            check_min_cppstd(self, 20)
+        if self.options.shared and self.options.static:
+            raise ConanInvalidConfiguration("Cannot set both shared and static libraries.")
```

Comment:
You should only provide `shared` option, instead of giving a way of providing both and then checking only one is provided.